contrib/buildrpm/buildrpmLibfabric.sh

```diff
@@ -0,0 +1,368 @@
+if [[ $# -lt 1 ]]; then
+  echo "$usage" 1>&2
+  exit 1
+fi
+
+#############
+# Print hello
+#############
+if [[ -n "$verbose" ]]; then
+cat <<EOF
+
+------------------------------------------------------------------------------
+====              Welcome to Libfabric RPM build script                   ====
+------------------------------------------------------------------------------
+You have specified this options:
+ $arguments
+------------------------------------------------------------------------------
+
+EOF
+  if [[ -n "$noop" ]]; then
+    echo "$0: --> Option -n was specified. I won't do anything."
+  fi
+fi
+
+################################
+# Check if tarball was specified
+################################
+export tarball="$1"
+
+if [[ ! -f "$tarball" ]]; then
+  echo "$0: FATAL: Can't find $tarball" 1>&2
+  exit 2
+fi
+if [[ -n "$verbose" ]]; then
+  echo "$0: --> Found tarball: $tarball"
+fi
+
+##############################
+# Get tarball name and version
+##############################
+export tardirname=$(basename "$tarball" | awk '{print substr ($0, 0, index($0, ".tar")-1)}')
+if [[ -z "$tardirname" ]]; then
+  echo "$0: FATAL: Cannot determine name from $tarball" 1>&2
+  exit 3
+fi
+if [[ -n "$verbose" ]]; then
+  echo "$0: --> Name of package: $tarball"
+fi
+
+export version=$(basename "$tardirname" | cut -d- -f2)
+if [[ -z "$tardirname" ]]; then
+  echo "$0: FATAL: Cannot determine version from $tarball" 1>&2
+  exit 4
+fi
+if [[ -n "$verbose" ]]; then
+  echo "$0: --> Version: $version"
+fi
+
+if [[ -n "$install_in_opt" ]]; then
+  prefix="/opt/libfabric/$version"
+  rpmbuild_options="$rpmbuild_options --define '_prefix $prefix'"
+  if [[ -n "$verbose" ]]; then
+    echo "$0: --> Setting RPM install path to: $prefix"
+  fi
+fi
+
+######################################
+# Try to unpack spec file from tarball
+######################################
+export tar_cmd=""
+export cp_cmd=""
+export rm_cmd=""
+
+if [[ -f "$specfile" ]]; then
+  if [[ -n "$verbose" ]]; then
+    echo "$0: WARNING: $specfile already exists and will be overwritten" 1>&2
+  fi
+fi
+
+if [[ -n "$unpack_spec" ]]; then
+  if [[ -z "$verbose" ]]; then
+    if [[ -z "$noop" ]]; then
+      tar -xf "$tarball"
+      cp -fp "$tardirname/$specfile" .
+      rm -rf "$tardirname"
+    fi
+  else
+    tar_cmd="tar -xvf \"$tarball\""
+    cp_cmd="cp -fpv \"$tardirname/$specfile\" ."
+    rm_cmd="rm -rfv \"$tardirname\""
+    if [[ -z "$noop" ]]; then
+      echo "$0: --> Extracting tarball"
+      eval "$tar_cmd"
+      echo "$0: --> Copying specfile"
+      eval "$cp_cmd"
+      echo "$0: --> Cleanup after extraction"
+      eval "$rm_cmd"
+    else
+      echo "$0: --> Extracting tarball using command:"
+      echo "$0: --> $tar_cmd"
+      echo "$0: --> Copying specfile with command:"
+      echo "$0: --> $cp_cmd"
+      echo "$0: --> Cleanup after extraction with command:"
+      echo "$0: --> $rm_cmd"
+    fi
+  fi
+fi
```

Comment:
Similar to the `verbose()` idea, you can also have another sub for executing (or not) commands.  E.g.,

```sh
doit() {
    verbose "Executing: $*"
    if test $noop -eq 0; then
        eval $*
        st=$?
        if test $st -ne 0; then
            echo "FATAL: command failed with status $st: $*"
            exit 1
        fi
    fi
}

doit tar -xf "$tarball"
doit cp -rf "$tardirname/$specfile" .
doit rm -rf "$tardirname"
# ...etc.
```

Note that checking the return value for the executed commands is fairly important -- you wouldn't want some intermediate command to fail, but then allow the script to continue.

Using a sub like this will considerably simplify the tar/cp/rm code block, above.